packages/client/lib/sync/fetcher/blockfetcher.ts

```diff
@@ -66,17 +66,24 @@ export class BlockFetcher extends BlockFetcherBase<Block[], Block> {
       `Requested blocks=${blocksRange} from ${peerInfo} (received: ${headers.length} headers / ${bodies.length} bodies)`
     )
     const blocks: Block[] = []
-    for (const [i, [txsData, unclesData]] of bodies.entries()) {
+    for (const [i, [txsData, unclesData, withdrawalsData]] of bodies.entries()) {
```

Comment:
I'm assuming referencing `withdrawalsData` on a pre-Shanghai block won't draw an error referencing an undefined object, right?  All the tests are passing so this appears to be the case and you explicitly check for `withdrawalsData` existing below.

Reply:
yes, it will come undefined